README.md

```diff
@@ -2,65 +2,57 @@
+
+**Example:** `sid:api.example.com/auth?uid=some-guid`
+
+### Wallet Compatibility
 
 Wallet compatibility relies on the implementation of the following requirements:
 
-- Scan QR code or retrieve Base64 authentication request
-- Decode Base64 string authentication request into:
-    ```JSON
-    {
-        "callbackUrl": "https://api.example.com/auth?connectionId=someConnection",
-        "nonce": "some-guid"
-    }
-    ```
+- Ability to scan QR code or retrieve Stratis ID URI
 - User or wallet should validate the callback url of the authentication request
-- Sign the nonce property value of the decoded request
-- _**POST**_ the signed message and public key to the callbackUrl property value of the decoded string
-    ```JSON
+- Sign the UID query string value of the request
+- _**POST**_ the JSON-encoded signed message and and public key to the callback URI, using HTTPS
+    ```
+    POST https://api.example.com/auth?uid=some-guid
+    Content-Type: application/json
     {
         "signature": "signed-message",
         "publicKey": "public-key"
     }
     ```
-___
-
-## Backend Implementation
-
-Backend compatibility relies on server side implementation of the following requirements:
-
-### Using Sockets
+    
+### dApp Compatibility
 
-- Connect to web socket with session or connection identifier
-  - Generate nonce (GUID) and callbackUrl on socket connection
-  - Store nonce, callbackUrl and timestamp
-  - Create and Base64 encode an auth request shown in [Wallet Implementation](#wallet-implementation)
-- Display QR code of Base64 encoded auth request
-- Implement public _**POST**_ endpoint at the callbackUrl that signatures will be sent to
-  - Example: `https://api.example.com/auth?connectionId=someConnection`
-  - With requests shown in [Wallet Implementation](#wallet-implementation)
-  - Validate signatures with public key and timestamp
-  - Generate auth token, remove auth request from store
-  - Pass auth token via socket to client
+dApp compatiblity relies on the implementation of the following requirements:
 
-### Using Polling
-
-Alternatively, polling can be used with additional backend steps:
-
-- Client makes an auth request to the server
-  - Server generates and returns an auth request with connectionId, nonce, timestamp and secret accessCode
-- Wallet signs and uses the callbackUrl to return a signature
-- Server validates signature, timestamp and connectionId, updating the data store with the results
-  - Throws away signature after use, not stored
-- Client polls API using connectionId and secret accessCode
-  - When found, generate and return auth token, deleting stored auth request records and access codes
+- Host a HTTPS endpoint that is used in the Stratis ID URI callback
+  - The endpoint must validate the structure and contents of the request body
+  - The endpoint must verify the signature and its contents
+- Generate a QR code containing a Stratis ID URI and display this to the user
```

Comment:
I think we should keep the details we had and maybe refactor to be clearer, the sockets/polling details are helping for projects that have no idea about that part and are looking for clear guidelines on implementation. 

Ripping it out basically says "we know how we're going to do it, figure the rest out on your own" to me

Reply:
We can include this information, although I think it should be in a separate document. This protocol doesn't have to be used for a dApp which is using websockets to authenticate a user. For example, you could build an extension to the full node that adds a callback endpoint, which triggers an event such as a transaction in the full node. This might be useful for a scenario where you want an application that alters the state of the blockchain or executes a method on a smart contract, if a user can prove they control a private key. That's just one idea.

Reply:
Separate doc is a happy medium I think, just so first time users have a quick idea how they could use it without trying to put the other pieces together themselves.

Reply:
I've added a document for a use case with web sockets. I've not used the original diagrams, as they include the database, which shouldn't be necessary to validate signatures. I think the timestamp stage which you originally specified could be important. The purpose of the timestamp would be to ensure old signatures aren't used. The UID _should_ be unique for every time it's used but we cannot realistically validate a signature that is received is not reused twice. However, maybe we replace this with a timestamp within the signature?

Reply:
Timestamp I think needs to be included somewhere for sure. Either privately in our DB, or included in the nonce that the wallet is going to signed. 

We should specify the connectionId should be a GUID specifically. Integers can be guessed and bad actors can just post signatures as they please to different UID based auth endpoints. Sneaky things can be attempted or just annoying UX things, being pushed an auth token for a different wallet address than mine.

Reply:
Not sure we should be mandating anything more than for it to be a unique string. Someone might decide a SHA hash works better for them, or a custom formed timestamp-based UID. If we want to define a public protocol we're better off making it as flexible as possible.